accounts/store_test.go

```diff
@@ -105,6 +107,140 @@ func assertEqualAccounts(t *testing.T, expected,
 	actual.LastUpdate = actualUpdate
 }
 
+// TestAccountUpdateMethods tests that all the Store methods that update an
+// account work correctly.
+func TestAccountUpdateMethods(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run("UpdateAccountBalanceAndExpiry", func(t *testing.T) {
+		store := NewTestDB(t)
+
+		acct, err := store.NewAccount(ctx, 0, time.Time{}, "foo")
+		require.NoError(t, err)
+
+		assertBalanceAndExpiry := func(balance lnwire.MilliSatoshi,
+			expiry time.Time) {
+
+			dbAcct, err := store.Account(ctx, acct.ID)
+			require.NoError(t, err)
+			require.EqualValues(t, balance, dbAcct.CurrentBalance)
+			require.WithinDuration(
+				t, expiry, dbAcct.ExpirationDate, 0,
+			)
+		}
+
+		// Get the account from the store and check to see what its
+		// initial balance and expiry fields are set to.
+		assertBalanceAndExpiry(0, time.Time{})
+
+		// Now, update just the balance of the account.
+		newBalance := lnwire.MilliSatoshi(123)
+		err = store.UpdateAccountBalanceAndExpiry(
```

Comment:
This applies to all tests:
I think we should also test how these db functions behave if an unknown id is passed into the functions.

Reply:
added